docs/langgraph/multi-agent.md

```diff
@@ -1,16 +1,308 @@
+  defaultTask: travelAdvisorTask,
+});
+
+// Test multi-turn conversation
+const config = { configurable: { thread_id: "conversation_1" } };
+
+let result = await multiTurnEntrypoint.invoke({
+  messages: [{ role: "user", content: "I want to plan a trip to Paris" }]
+}, config);
+
+// Continue the conversation
+while (true) {
+  const userInput = await interrupt("Please provide your next message:");
+  
+  result = await multiTurnEntrypoint.invoke({
+    messages: [{ role: "user", content: userInput }]
+  }, config);
+  
+  console.log("Assistant:", result.messages[result.messages.length - 1].content);
+}
+```
+
+## How to manage conversation history
+
+When building chatbots, one of the most important considerations is how to manage conversation history. Too much history can distract the model, while too little history can make the conversation feel impersonal.
+
+```typescript
+import { StateGraph, Annotation, START } from "@langchain/langgraph";
+import { BaseMessage, HumanMessage, AIMessage } from "@langchain/core/messages";
+
+// Define state
+const AgentState = Annotation.Root({
+  messages: Annotation<BaseMessage[]>({
+    reducer: (x, y) => x.concat(y),
+  }),
+});
+
+// Filter messages to keep only the most recent
+const filterMessages = (messages: BaseMessage[]) => {
+  return messages.slice(-1); // Keep only the last message
+};
+
+// Agent node that uses filtered messages
+const agent = async (state: typeof AgentState.State) => {
+  const { messages } = state;
+  const filteredMessages = filterMessages(messages);
+  
+  const response = await model.invoke(filteredMessages);
+  
+  return {
+    messages: [response],
+  };
+};
+
+// Build workflow
+const workflow = new StateGraph(AgentState)
+  .addNode("agent", agent)
+  .addEdge(START, "agent");
+
+const app = workflow.compile();
+```
+
+## How to add and use subgraphs
+
+Subgraphs allow you to build complex systems with multiple components that are themselves graphs. A common use case for using subgraphs is building multi-agent systems.
+
+### Add a node with the compiled subgraph
+
+```typescript
+import { StateGraph, Annotation } from "@langchain/langgraph";
+
+const SubgraphStateAnnotation = Annotation.Root({
+  foo: Annotation<string>, // note that this key is shared with the parent graph state
+  bar: Annotation<string>,
+});
+
+const subgraphNode1 = async (state: typeof SubgraphStateAnnotation.State) => {
+  return { bar: "bar" };
+};
+
+const subgraphNode2 = async (state: typeof SubgraphStateAnnotation.State) => {
+  // note that this node is using a state key ('bar') that is only available in the subgraph
+  // and is sending update on the shared state key ('foo')
+  return { foo: state.foo + state.bar };
+};
+
+const subgraphBuilder = new StateGraph(SubgraphStateAnnotation)
+  .addNode("subgraphNode1", subgraphNode1)
+  .addNode("subgraphNode2", subgraphNode2)
+  .addEdge("__start__", "subgraphNode1")
+  .addEdge("subgraphNode1", "subgraphNode2");
```

Comment:
We should actually need to add END.

```suggestion
  .addEdge("subgraphNode1", "subgraphNode2");
  .addEdge("subgraphNode2", END);
```

Reply:
This has also been corrected.
https://github.com/liam-hq/liam/pull/3491/commits/72a3e58e77b0ed8907a556e5ef6b911a4dc7b448